assets/environment-interpreter/common/etc/profile.d/0503_java.sh

```diff
@@ -0,0 +1,15 @@
+# shellcheck shell=bash
+# ============================================================================ #
+#
+# Setup Java
+#
+# ---------------------------------------------------------------------------- #
+
+if [[ -x "$FLOX_ENV/bin/java" ]]; then
+  export JAVA_HOME="$FLOX_ENV"
```

Comment:
fwiw nixpkgs packages the jdk with a shellHook that's pointing to `$out/lib/openjdk`, which in our case would be `$FLOX_ENV/lib/openjdk`. [1]
During builds, nixpkgs also sets `CLASSPATH` by scanning for jars in all dependency store paths under `$dependency/share/java/*.jar` [2].

I think it would be sensible to stick to nixpkgs idioms if we still want to present "dev" envrionemnts / manifest builds as an on-ramp to pure nix builds.


[1] https://github.com/NixOS/nixpkgs/blob/f92e02488663e3b290fde1640e53c9ccb010b00b/pkgs/development/compilers/openjdk/generic.nix#L569

[2] https://github.com/NixOS/nixpkgs/blob/f92e02488663e3b290fde1640e53c9ccb010b00b/doc/languages-frameworks/java.section.md?plain=1#L47

Reply:
Checked in about merging this for the release
- Re lib/openjdk: openjdk puts binaries in both bin and lib/openjdk, but other packages like graalvm only put binaries in bin, so setting to bin works in both cases
- Re CLASSPATH: we can maybe followup at some point to set that, but it would probably be better to spend some more time on the nixpkgs provided Java deps case if we go that route. It sounds if you're using Maven you don't need CLASSPATH set, so I'm not sure we need to set it if that would be an uncommon usage pattern. But we can get this fix into the release and come back to CLASSPATH if we want